File: backend/schemas/access_policy.py
from datetime import datetime

from backend.config import settings
from backend.schemas.base import CustomBaseModel


class AccessLevel(CustomBaseModel):
    id: int
    name: str
    description: str | None = None
    company_id: int
    is_active: bool = True
    created_at: datetime
    updated_at: datetime | None = None


class AccessLevelZone(CustomBaseModel):
    id: int
    access_level_id: int
    zone_id: int
    schedule_id: int | None = None
    out_of_hours_behavior: str = "deny"
    priority: int = 0


class AccessSchedule(CustomBaseModel):
    id: int
    name: str
    company_id: int
    timezone: str = settings.TIMEZONE
    config: dict = {}
    holiday_override_auto: bool = True
    is_active: bool = True
    created_at: datetime

File: frontend/src/graphql/queries/accessPolicy.ts
import { gql } from '@apollo/client';

export const ACCESS_LEVELS_QUERY = gql`
  query GetAccessLevels($companyId: Int!) {
    accessLevels(companyId: $companyId) {
      id
      name
      description
      companyId
      isActive
      createdAt
      updatedAt
      zoneAssignments {
        id
        accessLevelId
        zoneId
        scheduleId
        outOfHoursBehavior
        priority
      }
      userCount
    }
  }
`;

export const ACCESS_LEVEL_QUERY = gql`
  query GetAccessLevel($id: Int!) {
    accessLevel(id: $id) {
      id
      name
      description
      companyId
      isActive
      createdAt
      updatedAt
      zoneAssignments {
        id
        accessLevelId
        zoneId
        scheduleId
        outOfHoursBehavior
        priority
      }
      userCount
    }
  }
`;

export const ACCESS_SCHEDULES_QUERY = gql`
  query GetAccessSchedules($companyId: Int!) {
    accessSchedules(companyId: $companyId) {
      id
      name
      companyId
      timezone
      config
      holidayOverrideAuto
      isActive
      createdAt
    }
  }
`;
